### PR TITLE
Adds new evolutions for zombie soldiers (done; playtests would be nice)

### PR DESCRIPTION
**Pull request by I-am-Erk**
#### Summary
```SUMMARY: Content "Adds new evolutionary lines for zombie soldiers"```

#### Purpose of change
Currently zombie soldiers are threatening at the beginning of the game, but rapidly fall off as they do not evolve along with other zombies.

#### Describe the solution
This adds three new lines of zombie soldiers:

Black ops soldiers:
- These mix elements of shady zombies and runner zombies, creating a fast-moving hunt-by-night monster that eventually develops keen smell and is likely to, at some point during my play, cause me to pee my pants a little when it suddenly appears. I have not yet given the high level ones jumping; I might. (Edit: just noticed I forgot to bump bite damage on the Hunter-Killer up)

Acid-weapon soldiers 
- These need a better name. Both use a gun-type acid (like the corrosive zombie) that deals small amounts of hard to avoid damage.
- "snipers" have a medium-long range, slow-refire shot. It's long range for a zombie, about 20 tiles. In playtests they rarely see you soon enough to start shooting before you see them, but they do keep picking at you while you flee, which is neat.
- "support" have a spray/salvo weapon. It shoots three very inaccurate shots at close range, hitting 1/4 to 1/3 of the time, with a recharge of 4 turns. They're designed to counter melee, since they're pretty easy to kite at range, but if you come running at them they'll get a shot or two at you and your npc as you approach. The spray means they get a lot of collateral damage, which has been pretty cool in play tests as they smash cars and windows and hit NPC allies.
- Damage on these weapons still needs a lot of play testing. I've made both of them mix acid and cutting damage, for the bone dart and the acid covering it. The acid helps keep them relevant if you're wearing most high grade armours, but it's not likely to be a big deal. The cutting damage makes them hurt, but can be armoured against like anything. Snipers do less damage, because getting hit heavily at range would suck, and support does a decent chunk of it but at close range. Still needs further testing with various common armours but I'm getting pretty happy with it.

Kevlar zombies:
- these merge the skeleton and brute lines with armoured soldier or cop Z that have mutated. The first evo is durable and deals a pretty good chunk of slam damage. The second evo is smaller and less tough than a skeletal juggernaut, does similar damage, and is more resistant to blunt weapons... Kind of a halfway point to a skeletal juggernaut, it asks similar "whatcha gonna do, punk" questions but isn't quite the same degree of boss fight.

#### Describe alternatives you've considered

I could just make stronger versions of soldier zombies but that seems pretty dull.

The acid soldiers could probably instead use actual guns, but I prefer this being biological to having them remember their gear.

#### Additional context
I plan to give firefighters and cops a similar treatment.

I do not intend to have these spawn naturally at any time. They will, however, evolve into each other when done.

### CODE REVIEW

**Comment by darktoes** [comment]
The black op zombies sound great.

Acid sniper sound a bit rough, is there the possibility of something more along the lines of a bone sniper? Something that shoots bone spikes at high velocity? Support might be okay, but it sounds like it should be an upgrade from the spitter zombie. Maybe your support could use some sort of chemical warfare? Tossing gas grenades or something?

Tank zombies sound fun, but there already exists the armored zombie, which was apparently wearing power armor when they were killed. Perhaps they could be shifted in as an upgrade to the soldier zombie? For that matter, there's also the burner zombie (flamethrower) and a couple other zombies that are defined ingame but I've never actually seen spawn. Perhaps you could look at finding a place for them?

As another line, could I suggest a frogman-style underwater soldier zombie? Perhaps they could expel water at high pressure?

**Comment by WilliamtheCardboardBox** [comment]
Zombie acid snipers would be a little bit too much I'd much rather face actual guns rather than a spray of acid covering my entire body. I already hate acid zombies. Bullet proofing things is kinda boring since it's really all coming around to the tank at the end. And I like the idea of a bloodhound type black ops zombie which actually gives players a run for their money.

**Comment by thethunderhawk** [comment]
afaik evolutions are supposed to represent the blob changing existing zombies to give them new abilities. Idk if it makes sense for human soldiers who became zombies to “evolve” into zombified versions of more elite soldiers.

**Comment by Brian-Otten** [comment]
I love the new zombies, but got to agree i think the acid sniper might be a bit much just because of the sheer annoyance / extra micro factor. Am otherwise for the pull 100%

**Comment by AMurkin** [comment]
Wouldn't it be more sensible to just give them the abilities of zombie masters?
Not extraordinary: increased dodge, more HP, better night vision.

**Comment by aoae23** [comment]
In the same theme, zombie cops desperately need an evolution too.

**Comment by Fris0uman** [comment]
>Idk if it makes sense for human soldiers who became zombies to “evolve” into zombified versions of more elite soldiers.

This.
Maybe you could reflavor a little bit so it sounds more like classsic blob evolution and less like promotion.

Also acid sniper seems more like a very late evolution of acid zombie as it doesn't really fit the soldier theme I think.

**Comment by paulenka-aleh** [comment]
I like the idea of zombies being able to integrate body with their gun and knife and start shooting, at least it will make player to face bullets fired not from robots and not from NPCs. Not sure why blob wouldn't do that.
And it could really make my melee-oriented characters run at least in mid-game.

**Comment by I-am-Erk** [comment]
I sense an odd dichotomy between "shouldn't these be less soldier themed" with the black-ops line, and "these should be more soldier themed" with the acid line.

In truth I just picked two special mutant types, and made solider versions of them. I did pick shady because it seemed to fit nicely thematically, I picked acid because I wanted an acid variant among the groups I'm evolving.

There will also be a smoker firefighter, a necromancer firefighter, a boomer cop, and a grappler cop.

The sniper and support will be further altered, I don't plan to keep the current gun stats from the acid gun used by the corrosive zombie. I want them to have less burst fire and more dispersion, so they they can miss, especially the support zombie. Keep in mind for those really annoyed by acid, this is the bullet type of acid, not the pool of foot burning type.

It would be possible to just make the soldier upgrade into a faster soldier with higher dodge and other stats, and that's what the tank line will be like, but it is boring and sort of out of the design of the game to make that the primary Evo. Then it's just linear scaling like Oblivion: you get better armour, we make better monsters, nothing changes.

**Comment by Photoloss** [comment]
> I like the idea of zombies being able to integrate body with their gun and knife and start shooting, at least it will make player to face bullets fired not from robots and not from NPCs. Not sure why blob wouldn't do that.

Reminds me of the [Infested Marine](https://vignette.wikia.nocookie.net/starcraft/images/0/0d/InfestedTerran_SC2_Cncpt1.jpg/revision/latest?cb=20170203032055) from Starcraft a lot. Those are wearing power armour and we'd need a more fitting aesthetic anyway but a heavily mutated, slowly shambling powerhouse with tendrils firing their gun from the hip sounds a) horrific enough and b) not overly rage-inducing yet still very threatening to actually face in combat.

**Comment by anothersimulacrum** [comment]
Related to #28002?

**Comment by I-am-Erk** [comment]
Not that I was aware of, it's more about me being frustrated that evolution still isn't fixed, and following my usual "money where my mouth is" approach.

**Comment by Night-Pryanik** [comment]
> start shooting

We have a verbal agreement that zombies shouldn't be able to manipulate complex objects such as guns, with a very rare exception such as bio-operator (being even probably the only exception).

**Comment by Photoloss** [comment]
> We have a verbal agreement that zombies shouldn't be able to manipulate complex objects such as guns, with a very rare exception such as bio-operator (being even probably the only exception).

For future reference, what is the exact limit on this? Any old day1 zombie can grab **you** and hold you in place so can they grab random junk and use it as a club or is the concept of an inanimate tool completely alien to the Blob pilot? Would they be able to shakily point the barrel in your general direction and hold down the trigger on full auto? (obviously implying a complete inability to aim via sights, reload, deliberately switch firing modes or operate multi-switch safeties like on chainsaws)

**Comment by Night-Pryanik** [comment]
> so can they grab random junk and use it as a club

No, even this degree of manipulating things is too much. Grabbing you in place, punching you in face, throwing other creatures as they get in the way, biting you and other similar activities - that's the threshold most zombies can't cross.

**Comment by Photoloss** [comment]
> No, even this degree of manipulating things is too much. Grabbing you in place, punching you in face, throwing other creatures as they get in the way, biting you and other similar activities - that's the threshold most zombies can't cross.

Oh wow that's *really* limiting them. Is it the physical motor skills/precision or the concept of a tool that's stopping them? I.e. would the mutated "bone shooter" mentioned by a previous poster be feasible? Skeljug arms function as a kind of whip or flail, do those represent the upper limit of mutant dexterity or would a bone-based "rapier fencer" be possible?

**Comment by thethunderhawk** [comment]
If you’re looking for a lore-based answer, IMO the bit is that the blob doesn’t think of inanimate objects as useful. It wouldn’t occur to it to use a tool that isn’t part of itself. Maybe if the blob had a specific form designed for outside the box critical thinking (brain blob leaps to mind) some part of it could decide to emulate human tool use, but in general the blob wouldn’t spend the time or energy figuring out how to use human tools.

**Comment by Night-Pryanik** [comment]
@thethunderhawk said it totally right.

**Comment by DracoGriffin** [comment]
Shooting type zombies always seem a bit off; and tacking on acid just seems like a way to screw with the meta of the game rather than actual basis. Some of the ideas just literally feel like reskins which you might as well just do with tilesets instead of mechanics. More blob-interactive zombies feels the way to go; example is a hulk thrower (discussed in other PRs/Issues) --- launching other zombies and pieces of terrain/cars/etc at the player, which feels a bit more "natural".

Black Ops Zombie sounds like a great evolution, and the Smoker Firefighter (which being somewhat armored and probably much slower, but can provide "cover" to a horde) is another great idea.

Acid Sniper just doesn't sound right and really just seems to be focused on "well fuck ranged players because why not" rather than a better basis. Why would a sniper be shooting acid? Where did it come from? And this niche already feels a bit fulfilled with corrosive zombie. If the damage isn't enough for corrosive, then maybe that needs to be tweaked. Otherwise it just seems like the same zombie but ooh, it's got +2 to range!

Same with Acid Support.

Tank zombie is basically fulfilled with the Armored Zombie (which maybe can be added to later evo groups to make it a bit more frequent in military spawns as it is a bit rare); so what can be done to separate it a bit between Skeletal Juggernaut and Armored Zombie (both are slow "tank" types)? Could be a simple cop evolution but a stepdown in stats from Armored Zombie. "SWAT Zombie" or the like.

> a necromancer firefighter, a boomer cop, and a grappler cop.

Why a necromancer firefighter? Why not any other type, seems a bit strange for a firefighter to be a necromancer type? Additionally, I feel like Masters and Necromancers will/should evolve into Overlords. However, if you're just looking for armored variants, why not just stick with the bone shtick? "[Ossified/Skeletal] Necromancer".

Boomer and grappler cops... why? Again, this feels like a tileset thing rather than requiring a json entry. Or is it just meant to be "Armored" variants? "It's a fat doughnut eating cop... and became a boomer with a kevlar vest!"

**Comment by dissociativity** [comment]
How about dare I say, acid boomers?

**Comment by I-am-Erk** [comment]
@DracoGriffin Some of what you're saying is merely a problem with names; as I said in the OP, "acid sniper" and "acid support" are just the working titles I've given them. This isn't a soldier with a biological acid powered sniper rifle, it's a zombie in kevlar that's mutated down the corrosive zombie line and therefore has some traits of soldiers and some traits of corrosive zombies. For the sake of play variety, I made two types. One has a slow shooting, longer range acidic dart; the other has a bulk-shooting, inaccurate, short range acid dart. I chose these instead of acid vomit because puddles of acid are a huge pain in play, while acid "guns" are dealt with quickly and less of a nuisance for players. Neither are likely to be nearly as problematic as you have expressed here nor in discord. In fact if anything, I think currently I've lowballed them and will need to buff them significantly to make them a legitimate threat at the levels they spawn at.

The "sniper" is actually my favourite of these. Because of the limits of zombie threat detection, it doesn't normally shoot at you until you're pretty close by, which keeps it from obnoxiously plinking you from offscreen (its range also isn't nearly that long), but it has enough range that it can harm you as you try to flee, something the hordes generally lack. It does not hide behind a camo screen and shoot you from a hundred tiles away, it's a zombie. You'd do better to read the stat block than the WIP name of it.

The "support" is a lot of fun too IMO. It's very inaccurate, but its darts recharge quickly, so every couple turns it fires three shots in a spray. It's as likely to hurt your allies, your car, other zombies, and nearby buildings as it is to hurt you (in playtests it hits about 1/3 to 1/4 of the time so far). That's the intent. It has a max range of about 8 tiles, and an effective range of about 5, so this is far from a 'fuck you' to ranged players, and I can only assume you felt it was that without looking at the stats.

Your comments about tank zombies don't really address the issue, which is that evolution mechanics ask for ways to mutate zombies further. Soldier zombies don't mutate into power armoured zombies later in the game and it doesn't make logical sense for more power armours to spawn over time... although interestingly that's actually not far off the play balance end I'm going for. Again, what I'm describing here is what happens when you take a soldier or cop zombie wearing kevlar, and have them also grow bony or leathery plates and larger muscles like we see in other zombies.

>Why a necromancer firefighter? ...

Why *not* a necromancer firefighter / grappler cop? It's one of the known ways for zombies to mutate, so I am making a variant necromancer that happens to be wearing its old turnout coat and fire protective gear. For the sake of gameplay diversity it will not just be a carbon copy of the necromancer with firefighter stats; like the shady soldiers, I'll make it a bit unique to try to capture the infinite diversity of the blob (currently planning: much slower timer on necromancer ability, screecher's wail added in, aggressive rather than leadership role). "Boomer" cops are probably going to have boomer vomit but not be bloated and explode, for example.

I think all of this could be summarized thus: 
Stop thinking of this as a specialized soldier/firefighter/cop, and instead consider that these "profession" zombies mutate like every other zombie. They still have their gear, and it gives them some benefits that make them challenging to face. Rather than making a special mutant for every single professional zombie, I gave two variant types to each one and will also be making a more common "normal" type: tanks for cops and soldiers (plain zombies in kevlar -> tough zombies in kevlar -> brutes in kevlar) and fireproof zombies for firefighters (plain zombies in turnout gear -> tough zombies in turnout gear... you get it).

**Comment by DracoGriffin** [comment]
Great, I appreciate the time you took to list that with a bit more detail. I think the opening post would have benefited with some of those additional details as that's why the PRs received the overhaul in #24928 to help authors "sell their idea" rather than requiring further debate to clarify. Hence why I won't look further into the code (especially on monster design) unless the rationale is solid and explained well.

**Comment by I-am-Erk** [comment]
Thanks. I added some of that to the OP already. It's difficult to anticipate questions people will ask until they've asked them.

**Comment by Marrim** [comment]
Zombies with guns would be a bad idea because the gun itself has limited ammo and thus isn't worth the effort for the blob to adapt to using it--regardless of the blob's capability of doing so. There is simply not enough time for the blob to adapt to the gun before the gun becomes useless!

Not so with armor. In a sense armor is much closer to being part of the host. Zombies would not scavenge for it but if a zombie already happens to be wearing a thick kevlar shell... the blob does seem to base its development on the advantages of a particular host body. Something like a carapace to complement the armor and patch the vulnerable spots? Kevlar is an organic material, I don't see why the blob couldn't produce it biologically instead of keratin.

**Comment by Frost-wood** [comment]
Think of the blob as using the human as a spacesuit-the blob can’t handle the harsh energies of our dimension, so a body is what allows the blob to be active.

In my Zombie hollow PR, i had the zombie have no bones to be able to do a weak ranged attack by stretching.

There a few options for ranged attacks, like having the zombie spit out a piece of blob that grappels the player, or tries to infect and mutates the player.  A limpet mode attack might be interesting.

You could do a health or stamina draining ranged blob throw with just the jsons.

**Comment by thethunderhawk** [comment]
Rather than the spacesuit theory, I prefer to think of the blob’s subversion of organic life to be an adaptational tool and it’s expression of intelligence. Rather than leveraging inanimate objects to accomplish goals the way humans do, they leverage the animate. Instead of interacting with limbs and digits they interact at a biological and chemical level. 

Not that the spacesuit thing is necessarily wrong, but the blob *does* exist in pure form in our dimension. 

As for ranged attacks, there really are a million options that’d fit the lore. The “acid sniper” for one is IMO pretty reasonable, I just wonder how it will act mechanically and how that will affect the “feel” of the zombie faction. Certainly if it’s rare enough and far enough up the evolution tree to not be a problem till later in the play through I see no reason why it’d be a problem.

**Comment by Frost-wood** [comment]
It’s actually a spacesuit according to lore, i don’t know if the lore posts are still around from KA101, but the “blob” is a Chuthulu type intelligence controllling many subprime planes. 

The whole blob motto is adapt-it adapts to different conditions of the subprime planes.

Fungolid motto is transform- it transforms the environmont to one of its home planes.

Triffild motto is to dominate or conquer- they have many subspecies under there contol like a fedual system with different houses competing against each other.

**Comment by thethunderhawk** [comment]
That old lore forum stuff, the design doc, etc are all very dated, not exactly canon. I’m pretty sure Kevin is planning to release a new design doc. Anyway, this is all neither here nor there. I think as far as the scope of this PR goes, the lore is being handled just fine.

**Comment by I-am-Erk** [comment]
OK, these are done, and I've tested them out with an early/midgame character and found them all about where I want them to be. If someone wants to try them out with their own characters, that would be great.

**Comment by thethunderhawk** [comment]
What exactly (lorewise) causes Johnny The Soldier to zombify into a different sort of zombie than Jimmy-The-Civilian? His gear? His training? His somewhat above-average physique? 

I think it’s important that we aren’t just saying “soldiers sneak and shoot good, their zombies should be extra sneaky and shooty too”.

**Comment by I-am-Erk** [comment]
I've answered that above already. These are just corrosive/shady zombies, wearing kevlar, with somewhat more physically fit bodies to start from. They have somewhat different stats because corrosive and shady zombies are not meant to all be identical, we just don't yet have a system to allow every zombie to be different. The blob doesn't have some "shady zombie" template it applies to everyone. By giving these a bit of variation from the norm I can capture some of the differences of the zombie lines.

File: data/json/monstergroups/zombie_upgrades.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
+[
+  {
+    "type": "monstergroup",
+    "name": "GROUP_SOLDIER_UPGRADE",
+    "default": "mon_zombie_soldier_kevlar_1",
```
Comment (kevingranade) [inline data/json/monstergroups/zombie_upgrades.json]: This id doesn't exist.  You either failed to commit it, or it's supposed to be mon_zombie_soldier.

Comment (I-am-Erk) [reply data/json/monstergroups/zombie_upgrades.json]: I thought I caught all of those. Better make sure my repo is up to date

**Comment by RickySandbox** [comment]
Awesome additions! :+1: 

One of my most recent worlds spawned an Osprey crash site just 3 tiles from the southern side of my starting location. They slowly smashed their way out of the wreckage (sooo veeerrry slowly :rofl: ) and slaughtered the NPC and nearby wildlife, which gave me access to gear and food, but canceled my freshly acquired mission to find an artifact. :man_shrugging: :joy: 

Story aside, were these fresh recruits to the blob army added to the appropriate blacklist mods?  These stood out when I skimmed the list: mundane, no acid, no big, no explosive, no monsters.  I'm not familiar with mods, so I can't be certain how many should be adjusted.

**Comment by I-am-Erk** [comment]
I didn't touch any blacklist mods, those might need to be updated by someone interested in the subject.

**Comment by RickySandbox** [comment]
I'd be willing to look into it after I've experienced them all in-game. At the moment, I'm not willing to spoil _Muh surprises!_ by digging any deeper. I don't want to have someone's experience of C:DDA ruined because things didn't work as expected. :man_shrugging: 

@I-am-Erk The commit broke something you didn't touch and that doesn't interest you enough to work on. Forget about it. Keep coming with the awesome additions! :heart_eyes:

**Pull request merged**